crates/ark/src/lsp/symbols.rs

```diff
@@ -33,6 +31,30 @@ use crate::treesitter::BinaryOperatorType;
 use crate::treesitter::NodeType;
 use crate::treesitter::NodeTypeExt;
 
+fn new_symbol(name: String, kind: SymbolKind, range: Range) -> DocumentSymbol {
+    DocumentSymbol {
+        name,
+        kind,
+        detail: None,
+        children: Some(Vec::new()),
+        deprecated: None,
+        tags: None,
+        range,
+        selection_range: range,
+    }
+}
+
+fn new_symbol_node(
+    name: String,
+    kind: SymbolKind,
+    range: Range,
+    children: Vec<DocumentSymbol>,
+) -> DocumentSymbol {
+    let mut symbol = new_symbol(name, kind, range);
+    symbol.children = Some(children);
+    symbol
+}
```

Comment:
One thing that I really like from the Biome group is the `with_<optional_feature>()` pattern

```rust
fn with_children(mut self, children) -> Self {
  self.children = children;
  self
}
```

So it becomes

```rust
new_symbol(symbol, kind, range).with_children(children)
```

It would require a `DocumentSymbolExt` trait so this is fine if you want to keep it as is, just thought id point it out as a pattern to keep in mind

Reply:
That does seem like a nice pattern!